programming-extensions/programming-extension-dataspaces/src/main/java/org/objectweb/proactive/extensions/dataspaces/vfs/selector/FileSelector.java

```diff
@@ -204,13 +210,63 @@ private PathMatcher asPathMatcher(String pattern) {
         if (!pattern.startsWith(PREFIX_GLOB_PATTERN) && !pattern.startsWith(PREFIX_REGEX_PATTERN)) {
             pattern = PREFIX_GLOB_PATTERN + pattern;
         }
+        if (pathMatcherCache.containsKey(pattern)) {
+            return pathMatcherCache.get(pattern);
+        }
+
+        PathMatcher matcher = fileSystem.getPathMatcher(pattern);
+        pathMatcherCache.put(pattern, matcher);
+        return matcher;
+    }
 
-        return fileSystem.getPathMatcher(pattern);
+    private String convertPathForAntPathMatcher(Path path, String targetSeparator) {
+        String convertedPath = path.toString().replace("\\", targetSeparator).replace("/", targetSeparator);
+        if (convertedPath.startsWith(targetSeparator)) {
+            convertedPath = convertedPath.substring(1);
```

Comment:
Are you sure that targetSeparator is always of length 1? Can we do `convertedPath.substring(targetSeparator.length());` ?